md4c/md4c.c

```diff
@@ -3497,6 +3535,74 @@ md_resolve_links(MD_CTX* ctx, const MD_LINE* lines, int n_lines)
     return 0;
 }
 
+static int
+md_is_inline_wikilink_spec(MD_CTX* ctx, int opener_index, int closer_index)
+
+{
+    MD_MARK* opener = &ctx->marks[opener_index];
+    MD_MARK* closer = &ctx->marks[closer_index];
+
+    int delim_index = opener_index;
+    MD_MARK* delim = &ctx->marks[delim_index];
+
+    /* No content; ignore. */
+    if (opener->end == closer->beg)
+        goto not_found;
+
+    while(delim_index < closer_index) {
+
+        /* The wiki link is ignored if the `|` delimiter is used, but without a
+         * label. TODO Unsure if this is the desired behavior. How about
+         * removing the `|` from the output instead, and keeping the link? */
+
+        /* The wiki link is also ignored if the delimiter is used, but without
+         * a target. */
+
+        if(delim->ch == '|' && (delim->end == closer->beg || delim->beg == opener->end)) {
+            goto not_found;
+        } else if(delim->ch == '|') {
+            opener->end = delim->beg;
+            break;
+        }
+
+        delim_index++;
```

Comment:
This scanning for the `|` likely opens the gates to quadratic behavior for `[[[[[[[[foo|bar]]]]]]]]....`.